src/components/Header/Header.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { TodoMethod, TodosContext } from '../../context/ToDosContext';
+import cn from 'classnames';
+
+export const Header: React.FC = () => {
+  const [title, setTitle] = useState('');
+
+  const { todos } = useContext(TodosContext);
+  const methods = useContext(TodoMethod);
+
+  const field = useRef<HTMLInputElement>(null);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const trimmedTitle = title.trim();
+
+    if (trimmedTitle) {
+      methods.addTodo(trimmedTitle);
+      setTitle('');
+      field.current?.focus();
+    }
+  };
```

Comment:
Could be wrapped in a `useCallback` for improved performance

Reply:
Done!